bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Display.java

```diff
@@ -1527,6 +1579,19 @@ public static Display getDefault () {
 	}
 }
 
+/**
+ * Returns an executor that will execute any submitted runnable in the user-interface thread, regardless of the calling thread
+ * @return an {@link Executor} that executes in the user-interface thread.
+ * @since 3.120
+ */
+public static Executor executor() {
```

Comment:
```suggestion
public static Display executor() {
```

Reply:
Looking on https://github.com/eclipse-platform/eclipse.platform.ui/pull/51 I wonder if the javadoc should mention, that the returned executor is not guaranteed to work if disposed by other code.

Also this would mean, the returned executor shouldn't be cached.

Reply:
Disposal is a good topic to mention here, I can think about that we check if the display is disposed and then simply *not* execute, this might be what most users actually want, but is a bit contradicted to the API of executor.

Beside that if we want to follow the Executor contract, I would simply throw `RejectedExecutionException` in that case, I think most of the time where the Display is disposed there is either something really wrong, or the system is shutting down, in both cases I won't mind to get a proper exception.

Reply:
"Not execute" would hide programming errors. 
Display will throw "Device disposed" anyway, do we really need extra handling here? 
And yes, I assume in most cases that shouldn't be a problem, but it would make sense to add a note that there **is** a possibility that the executor will not always work.

Reply:
> Display will throw "Device disposed" anyway

Yes, but that's a bit SWT specific here so I think throwing `RejectedExecutionException` with the SWT error as a cause would be suitable as it is a declared exception of the api contract:

> @throws RejectedExecutionException if this task cannot be accepted for execution

e.g. there are executors that only allow to have a fixed bounded list of pending executions ... I think this is a similar case here.

Reply:
About the return type I think `Executor` would be really better here to be less specific than possible and people not tend to accidently call sync/async exec for example?

Reply:
Why shouldn't they be able to do that? At the end, it's a Display that implements Executor now. Or should it not implement but delegate? But then the code would not be so easy to use as proposed.

Reply:
I think it should implement it to prevent additional roundtripps if you have a display (e.g. from a widget), but here the method is to get an `Executor` (and not a `Display`) that is we should not expose the implementation it self here but if we think its useful of course both would work. I just try to keep return values as open as necessary, e.g. returning Collection instead of List even though internally maybe an ArrayList is used.

Reply:
> would be a much more explicit and more powerful method

I could even add both ;-)

I just find this to verbose and trying to tell to much about internals, so following that we must rename `getCurrent()` -> `getCurrentOrNullIfNotUIThread()` and `getDefault()` -> `getDefaultOrCreateAndMakeCurrentThreadUIThread()` ...

But as mentioned before, if its just the name I could name it as you like ...

Reply:
> I could even add both ;-)

As usual with API, it's not the more the better, it's more the opposite. So let's try to only add what seems best.

> I just find this to verbose

I like verbose stuff, they help me in not having to read the documentation. However, any proposal to a shorter and as explicit name would be welcome. The name I suggest here is the one I thought of while reading the examples because I believe it makes everything clearer on consumer side.

> trying to tell to much about internals,

For the display, such details must matter to consumer. Someone who consumes a Display needs to get a sense of which one(s) they're using and get a sense that maybe `Display.executor()` is not the API they want because it's not necessarily the best display accessor to use in their context. We don't want things that work "magically", we want API that make user write good code. For this class, I believe exposing a bit more about the behavior in the method name allows that better.

Reply:
Well there is not much magic and thats why I have added the javadoc do clarify it do you think the javadoc is missing someting? At least for me the javadoc is more normative than any method name.

Reply:
> At least for me the javadoc is more normative than any method name.

People don't read doc. Naming is one of the most important part of software development (cf "Clean Code" by Bob Martin). Javadoc mustn't be a substitution to good naming.